### PR TITLE
PDA emoji

### PR DESCRIPTION
**Pull request by ghost**
allemoji

they're invoked by encapsulating their names in colons, right now you have:

- happy
- sad
- angry
- confused
- rolling_eyes
- pensive
- joy
- noface
- gun
- ok_hand
- middle_finger
- thinking
- thumbs_up
- thumbs_down
- tada
- rocket_ship
- heart

there's some bonus hidden emoji that I'm too lazy to keep adding to the picture but as of this edit I'm not adding any more for this PR

lynch me

:cl:
 * rscadd: You can now use emoji to more aptly express yourself in PDA messages.

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
Based

**Comment by Kharkov72** [comment]
time to mass pda 🤔 to everyone after stealing the HoS's PDA

**Comment by Lokiamis** [comment]
Newscaster emojicracy when

**Comment by ShiftyRail** [comment]
Make mimes unable to type anything in their PDA other than those emojis and a few selected punctuation signs

File: code/modules/emoji/emoji.dm [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/proc/emoji_parse(text)
```
Comment (ShiftyRail) [inline code/modules/emoji/emoji.dm]: I think it'd be a lot better to just use `findtext(text, ":thinking;")` and similar directly

Comment (ghost) [reply code/modules/emoji/emoji.dm]: I had to think about it for a moment but that would definitely solve some issues with this parsing algo (and be much smaller as a side effect). I'll rewrite it in a bit when I get home to findtext() iterate over a static emoji list.

**Comment by ghost** [comment]
> Make mimes unable to type anything in their PDA other than those emojis and a few selected punctuation signs

Already plan on doing it in a seperate PR.

**Comment by tuypo1** [comment]
I hate the idea of emoijs in regular pdas but making mimes able to use only emojis sounds like a great idea

**Review by ShiftyRail — APPROVED** [review]
My review is requested but I don't know shit about `asset` datums, I think either Damian, PJB, or Nexis would be more informed on the matter
:+1: regardless

**Comment by DamianX** [comment]
The assets are sent to the client before opening the PDA UI. Does this mean someone who has never opened the PDA wouldn't be able to load them within goonchat?

**Comment by ghost** [comment]
Yeah I realized that, when I get a chance I'll make it so they're downloaded when you receive a message as well.

**Comment by DamianX** [comment]
It might be easier to just send them when the client connects.

**Comment by ghost** [comment]
That was my original thought but I was concerned it would get called out for being inefficient. I'll just do that, then.

**Comment by ghost** [comment]
>Secret repo conflicts

fuck

**Review by DamianX — APPROVED** [review]
Seems OK. I guess we'll see in which unpredictable way this will fail.

**Comment by ShiftyRail** [comment]
I'll fix the secret repo too

**Pull request merged**